ocaml/libs/http-lib/http_svr.ml

```diff
@@ -359,11 +359,13 @@ let request_of_bio_exn ~proxy_seen ~read_timeout ~total_timeout ~max_length bio
     proxy |> Option.fold ~none:[] ~some:(fun p -> [("STUNNEL_PROXY", p)])
   in
   let open Http.Request in
-  (* Below transformation only keeps one value per key, whereas
-     a fully compliant implementation following Uri's interface
-     would operate on list of values for each key instead *)
   let kvlist_flatten ls =
-    List.map (function k, v :: _ -> (k, v) | k, [] -> (k, "")) ls
+    (* Uri.query splits the value string into several if they are separated
+       with commas. Like this: "?k=v1,v2,v3" -> [("k", ["v1";"v2";"v3"])]
+       This function concatenates these back. It will not concatenate values
+       entered for duplicate keys, as these will be separate tuples:
+       "?k=v1,v2,v3&k=v4" ->  [("k", ["v1"; "v2"; "v3"]); ("k", ["v4"])] *)
+    List.map (fun (k, vs) -> (k, Astring.String.concat ~sep:"," vs)) ls
```

Comment:
This works, although you don't necessarily have to use `Astring` for this.
There is `StringLabels.concat ~sep` in the standard library if you want a labeled version.

Otherwise looks correct.

Reply:
Indeed, `String.concat ","` is the most common of the three in xapi